mmdet/models/dense_heads/detr_head.py

```diff
@@ -529,43 +440,17 @@ def _get_targets_single(self, cls_score: Tensor, bbox_pred: Tensor,
         return (labels, label_weights, bbox_targets, bbox_weights, pos_inds,
                 neg_inds)
 
-    # over-write because img_metas are needed as inputs for bbox_head.
-    def loss(self, x: Tuple[Tensor], batch_data_samples: SampleList) -> dict:
-        """Perform forward propagation and loss calculation of the detection
-        head on the features of the upstream network.
-
-        Args:
-            x (tuple[Tensor]): Features from the upstream network, each is
-                a 4D-tensor.
-            batch_data_samples (List[:obj:`DetDataSample`]): The Data
-                Samples. It usually includes information such as
-                `gt_instance`, `gt_panoptic_seg` and `gt_sem_seg`.
-
-        Returns:
-            dict: A dictionary of loss components.
-        """
-        batch_gt_instances = []
-        batch_img_metas = []
-        for data_sample in batch_data_samples:
-            batch_img_metas.append(data_sample.metainfo)
-            batch_gt_instances.append(data_sample.gt_instances)
-
-        outs = self(x, batch_img_metas)
-        loss_inputs = outs + (batch_gt_instances, batch_img_metas)
-        losses = self.loss_by_feat(*loss_inputs)
-        return losses
-
     def loss_and_predict(self,
-                         x: Tuple[Tensor],
+                         hidden_states: Tuple[Tensor],
                          batch_data_samples: SampleList,
                          proposal_cfg: Optional[ConfigType] = None) \
-            -> Tuple[dict, InstanceList]:
+            -> Tuple[dict, InstanceList]:  # TODO: ?
```

Comment:
What TODO?

Reply:
This `TODO` is to remind me to change the redundant parameters here, and I have submitted the modified version. Thank you very much for the correction.